### ISSUE TITLE
Plugin should use dynamic metric catalog

### ISSUE THREAD

**Issue by mkleina** [intelsdi-x/snap-plugin-collector-ceph#26]
Plugin should build metric catalog dynamically, based on Ceph socket data.
Current hardcoded metric list can be incompatible with different versions of Ceph / different setup configurations.

**Comment by IzabellaRaulin** [comment]
@mkleina, I agree - this is **how this plugin worked before** (for sure for commit 8d747fc9ca227f6b3f74fba2c5d7dc8f1c0feedb). 
Hardcoded metric list allows load plugin without global configuration (so there is no provided path to ceph.socket). 
However, in such cases when there is an incompatibility in receiving the list of metrics from Ceph, I suppose your proposal might be better option.
